### PR TITLE
AQI-3743 SDK: Analyte and Test Group Part 1

### PR DESCRIPTION
**Pull request by DarshanaGandhi-AI**

### CODE REVIEW

File: src/Aquarius.ONE.ClientSDK/ClientSDK.cs [hunk 1]
```diff
@@ -9,6 +9,7 @@
 using ONE.Enterprise.Core;
 using ONE.Enterprise.Report;
 using ONE.Enterprise.Twin;
+using ONE.Operations.Sample;
```
Comment (KentMiller) [inline src/Aquarius.ONE.ClientSDK/ClientSDK.cs:12]: I don't think this using needs to be added, I viewed the full file and it already exists.

File: src/Aquarius.ONE.ClientSDK/Utilities/CacheExceptions.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace ONE.Utilities
+{
+    public class CacheExceptions
```
Comment (KentMiller) [inline src/Aquarius.ONE.ClientSDK/Utilities/CacheExceptions.cs:7]:
```suggestion
    Should mark the class as static.
```

File: src/Aquarius.ONE.ClientSDK/Operations/Sample/SampleApi.cs [hunk 3]
```diff
@@ -69,7 +72,245 @@ public async Task<bool> UpdateActivitiesAsync(List<Activity> activities)
...
         }
 
+        public async Task<bool> CreateAnalyteAsync(Analyte analyte)
+        {
+            var watch = System.Diagnostics.Stopwatch.StartNew();
+
+            var requestId = Guid.NewGuid();
+            var endpoint = $"/operations/sample/v1/analyte";
+
+            var json = JsonConvert.SerializeObject(analyte, new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
```
Comment (KentMiller) [inline src/Aquarius.ONE.ClientSDK/Operations/Sample/SampleApi.cs:82]: Could create a class instance of the JsonSerializerSettings and use it throughout the class when it's needed. Should also add the CamelCaseResolver.

File: src/Aquarius.ONE.ClientSDK/Operations/Sample/SampleCache.cs [hunk 4]
```diff
@@ -1,4 +1,7 @@
-﻿using System;
+﻿using Newtonsoft.Json;
```
Comment (KentMiller) [inline src/Aquarius.ONE.ClientSDK/Operations/Sample/SampleCache.cs:1]: Not sure if it's just the way the code is shown in this review but it looks like there are duplicate using statements.

**Review by AQI-DanG — CHANGES_REQUESTED** [review]
Please use protobuf methods where available for communicating with the apis.

File: src/Aquarius.ONE.ClientSDK/Common/Logbook/LogbookCache.cs [hunk 5]
```diff
@@ -94,7 +94,7 @@ public async Task<bool> LoadLogbooksAsync(string operationId = "")
         public async Task<bool> LoadEntriesByLogbookAsync(string logbookId, DateTime startDate, DateTime endDate)
         {
             if (!Logbooks.ContainsKey(logbookId))
-                return ErrorResponse(UnloadedException(logbookId), false);
+                return ErrorResponse(CacheExceptions.UnloadedException("Logbook", logbookId), false);
```
Comment (JonathanRobertson-AQI) [inline src/Aquarius.ONE.ClientSDK/Common/Logbook/LogbookCache.cs:97]: Was this change to share error handling across areas of the SDK?

Comment (DarshanaGandhi-AI) [reply src/Aquarius.ONE.ClientSDK/Common/Logbook/LogbookCache.cs:97]: yes, it is the class to share error handling across the all Cache classes

File: src/Aquarius.ONE.ClientSDK/Aquarius.ONE.ClientSDK.csproj [hunk 6]
```diff
@@ -5,7 +5,7 @@
 	  <!--For Release packages update the VersionPrefix and remove any VersionSuffix-->
 	  <VersionPrefix>13.15.0</VersionPrefix>
 	  <!--For Feature packages update the VersionSuffix with the Jira ticket number-->
-	  <VersionSuffix></VersionSuffix>
+	  <VersionSuffix>prerelease-1</VersionSuffix>
```
Comment (KentMiller) [inline src/Aquarius.ONE.ClientSDK/Aquarius.ONE.ClientSDK.csproj:8]: Not a huge deal but like the comment states, the VersionSuffix value for pre-release versions should contain the Jira ticket number. Something like aqi-3743-01

**Review by KentMiller — APPROVED** [review]

File: src/Aquarius.ONE.ClientSDK/Aquarius.ONE.ClientSDK.csproj (hunk 6 above)
Comment (JonathanRobertson-AQI) [reply src/Aquarius.ONE.ClientSDK/Aquarius.ONE.ClientSDK.csproj:8]: I'd prefer to keep this out of the public NuGet, now that we're using this. I'd vote for changing the comment to not include a Jira ticket number

**Review by JonathanRobertson-AQI — APPROVED** [review]

File: src/Aquarius.ONE.ClientSDK/Aquarius.ONE.ClientSDK.csproj (hunk 6 above)
Comment (AQI-DanG) [reply src/Aquarius.ONE.ClientSDK/Aquarius.ONE.ClientSDK.csproj:8]: Agreed, when we moved these packages to public nuget we stopped using jira ticket numbers although that perhaps was not communicated.

**Review by AQI-DanG — APPROVED** [review]

File: src/Aquarius.ONE.ClientSDK/Aquarius.ONE.ClientSDK.csproj (hunk 6 above)
Comment (DarshanaGandhi-AI) [reply src/Aquarius.ONE.ClientSDK/Aquarius.ONE.ClientSDK.csproj:8]: I already did that earlier but as per Jon it is public facing so I did change to prerelease-1. @JonathanRobertson-AQI  correct?

**Pull request merged**